docs/README.md

```diff
@@ -1,16 +1,42 @@
-# Developer-local docs build
+# Building documentation locally
+
+## Building in an existing envoy development environment
+
+If you have an existing envoy development environment, you should have the necessary dependencies
+and requirements and be able to build the documentation directly.
 
 ```bash
 ./docs/build.sh
 ```
 
-The output can be found in `generated/docs`. By default configuration examples are going to be validated during build. 
+The output can be found in `generated/docs`. By default configuration examples are going to be validated during build.
 To disable validation, set `SPHINX_SKIP_CONFIG_VALIDATION` environment variable to `true`:
 
 ```bash
 SPHINX_SKIP_CONFIG_VALIDATION=true docs/build.sh
 ```
 
+## Using the Docker build container to build the documentation
+
+If you *do not* have an existing development environment, you may wish to use the Docker build
+image that is used in continuous integration.
+
+This can be done as follows:
+
+```
+./ci/run_envoy_docker.sh 'ci/do_ci.sh docs'
+```
+
+To use this method you will need a minimum of 4-5GB of disk space available to accomodate the build image.
+
+# Creating a Pull Request with documentation changes
+
+When you create a Pull Request the documentation is rendered by CircleCI.
+
+This can be viewed by clicking "Details" in "ci/circleci: docs" check. From there click "ARTIFACTS", and then
+"generated/docs/index.html".
```

Comment:
@itayd im wondering if we could add a trigger on the repo bot so that

- whenever a commit is added that makes changes to `/docs/` it messages:
   - the url eg (https://384272-65214191-gh.circle-artifacts.com/0/generated/docs/index.html)
   - from the circleci API (~eg https://app.circleci.com/pipelines/github/envoyproxy/envoy/34403/workflows/33eaf7f9-214f-4991-aeff-620047b76ea0/jobs/384272)

Reply:
This would be awesome.

Reply:
how would i know the docs have been rebuilt? please create an issue for that.

Reply:
cool, will do